### PR TITLE
Local development enhancements

### PR DESCRIPTION
**Pull request by tekante**
## Description

Further enhancements for local development and addition of successful project creation via ability to run `yarn build` and `yarn storybook:build` within the project. This also has changes to test for successful project initiation in CI. See https://github.com/phase2/outline/actions/runs/4846186198 for example of a full successful run.

This builds on work from branch `feature/docs-and-init-dx` so I'm setting that as the target at the moment to make it easier to review what has changed since then.

Once #380 merges this can be targeted at the `next` branch for merging though note that the changes in #385 are needed for full successful CI.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Visual Testing
- [ ] Automated Testing
- [ ] Accessibility Testing

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -123,6 +118,54 @@ jobs:
       - name: Build Storybook
         run: yarn storybook:build
 
+  # Test initialization of a new project
+  build_project:
+    name: Initialize and Test Build of Project Codebase
+    runs-on: ubuntu-latest
```
Comment (mike-potter) [inline .github/workflows/build.yml:123]: Are we going to run out of Github action minutes using their runner here instead of `self-hosted`?

**Review by bardleb — APPROVED** [review]

**Review by shaal — APPROVED** [review]
@tekante thank you!

File: packages/outline-templates/default/scripts/link-for-local-dev.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/**
+ * Takes an argument of the path to a directory containing package directories to link for
+ * local development. Will register them all for local development in the current project.
+ */
```
Comment (himerus) [inline packages/outline-templates/default/scripts/link-for-local-dev.js:4]: Now that I'm looking through things, I'm thinking running `yarn turbo dev` does this essential same linking of the local packages. 

I've not used the mode but @grshane reminded me it was a thing.

Comment (himerus) [reply packages/outline-templates/default/scripts/link-for-local-dev.js:4]: So I'm pondering if we can be running `yarn turbo dev` and then run the CLI commands in a similar way.. .

**Review by himerus — COMMENTED** [review]
I think we may have some other options here. I've really got to dive further into the details here and what turborepo should be doing for us by default.

File: .github/workflows/build.yml (hunk 1 above)
Comment (tekante) [reply .github/workflows/build.yml:123]: I've confirmed that builds from this repo aren't counting against our action minutes allocation.

File: packages/outline-templates/default/scripts/link-for-local-dev.js (hunk 2 above)
Comment (tekante) [reply packages/outline-templates/default/scripts/link-for-local-dev.js:4]: This script is intended for use in a generated project. I don't think `yarn turbo dev` would apply in that context. Am I overlooking something?

Comment (himerus) [reply packages/outline-templates/default/scripts/link-for-local-dev.js:4]: Derp. In my head this was for Outline local dev.

Comment (himerus) [reply packages/outline-templates/default/scripts/link-for-local-dev.js:4]: This is what I was thinking of where `yarn turbo run dev` would possibly serve the same purpose as what's going on there. https://github.com/phase2/outline/actions/runs/5046433815/jobs/9052038359#step:7:15

It's worth some digging later.

**Pull request merged**